editor/filesystem_dock.cpp

```diff
@@ -2871,8 +2871,21 @@ void FileSystemDock::drop_data_fw(const Point2 &p_point, const Variant &p_data,
 		// Moving resource.
 		Ref<Resource> res = drag_data["resource"];
 		String to_dir;
-		bool favorite;
-		_get_drag_target_folder(to_dir, favorite, p_point, p_from);
+		if (p_from == tree) {
+			// _get_drag_target_folder() returns the wrong value at the edges of the TreeItem, so we use this approach instead.
+			TreeItem *ti = tree->get_item_at_position(p_point);
+			if (!ti) {
+				return;
+			}
+			to_dir = ti->get_metadata(0);
+			if (!to_dir.ends_with("/")) {
+				to_dir = to_dir.get_base_dir();
+			}
+		} else {
+			bool favorite;
+			_get_drag_target_folder(to_dir, favorite, p_point, p_from);
+		}
```

Comment:
You should make this into a method.

Reply:
I thought I did, actually. Must've forgot, I'll update it and push it right away, thank you!